torch/_functorch/aot_autograd.py

```diff
@@ -2649,12 +2775,15 @@ def create_aot_dispatcher_function(
     # Check flat_args to see if they're already fake.  If so, use that fake
     # mode instead.
 
+    fake_mode_init = False
     for x in flat_args:
         if isinstance(x, FakeTensor):
             fake_mode = x.fake_mode
             shape_env = fake_mode.shape_env
+            fake_mode_init = True
             break
-    else:
+
+    if not fake_mode_init:
```

Comment:
why?

Reply:
The original code looks wrong to me. The if .. else .. conditions were not indented.

Reply:
The original code is correct, this is a for .. else .. statement !
They are very convenient to do exactly this kind of code pattern without having to have a boolean flag around!

Reply:
Wow. TIL.